apps/sertra/pages/server/[id].tsx

```diff
@@ -0,0 +1,206 @@
+import { ServerListingDetailData, ServerListingData } from "../../api/models";
+import { PropsWithChildren } from "react";
+import styles from "../../components/ServerDetail.module.css";
+import { ApiURLs } from "../../api/urls";
+
+export async function getStaticPaths() {
+  const res = await fetch(ApiURLs.ServerList);
+  const listings = await res.json();
+
+  const paths = listings?.results.map((listing: ServerListingData) => ({
+    params: { id: listing.id },
+  }));
+
+  return { paths, fallback: "blocking" };
+}
+
+export async function getStaticProps(context: any) {
+  const id = context.params.id;
+  const res = await fetch(ApiURLs.ServerDetail(id));
+  const data = await res.json();
+
+  return {
+    props: {
+      detail_listing: data,
+    },
+    revalidate: 10,
+  };
+}
+
+interface ServerModeProps {
+  isPvP: boolean;
+}
+const ServerMode: React.FC<PropsWithChildren<ServerModeProps>> = ({
```

Comment:
This component (and `ServerPasswordProps` below) seems to be copypasta from server listings view. Usually the way would be to make them separate, reusable components. But as it happens, my current PR introduces [changes to those components](https://github.com/thunderstore-io/thunderstore-ui/pull/492/files), so maybe the separation should be done after that PR is (hopefully) merged.

Reply:
Rebased on master and added a commit that separates the listing attributes to it's own file. And then removed the copypasta.

Reply:
Looks good, except I'd capitalize the name of the new component file. The code base is currently inconsistent in naming files (among other things), but I'd assume we're going to end up using PascalCase.